### PR TITLE
Fix addMedCon empty input bug

### LINKED ISSUE #136: Bug: Unable to add/delete medical conditions when state of filtered list is different
After using find/findMedCon/findNric/listPrio commands to change the state of the filtered list, the user will be unable to add/delete med con for another user even if their name does not appear in the filtered list

### PR DESCRIPTION
**Pull request by hweinian**
fixes #136

### CODE REVIEW

File: src/main/java/seedu/address/logic/parser/AddMedConCommandParser.java [hunk 1]
```diff
@@ -45,10 +45,14 @@ public AddMedConCommand parse(String args) throws ParseException {
         // Parse all MedCon values and add them to a set
         Set<MedCon> medCons = new HashSet<>();
         for (String medConStr : argMultimap.getAllValues(PREFIX_MEDCON)) {
-            if (medConStr.length() > 45) {
+            if (medConStr.isEmpty()) {
+                throw new ParseException(String.format(MESSAGE_INVALID_COMMAND_FORMAT,
+                        AddMedConCommand.MESSAGE_USAGE));
+            } else if (medConStr.length() > 45) {
                 throw new ParseException(MESSAGE_CONSTRAINTS_LENGTH);
```
Comment (yadobler) [inline src/main/java/seedu/address/logic/parser/AddMedConCommandParser.java:52]: good guard clauses

File: src/main/java/seedu/address/logic/parser/DelMedConCommandParser.java [hunk 2]
```diff
@@ -51,11 +51,16 @@ public DelMedConCommand parse(String args) throws ParseException {
         // Parse all MedCon values and add them to a set
         Set<MedCon> medCons = new HashSet<>();
         for (String medConStr : argMultimap.getAllValues(PREFIX_MEDCON)) {
-            if (medConStr.length() > 45) {
+            if (medConStr.isEmpty()) {
+                logger.warning("Medical condition is empty.");
+                throw new ParseException(String.format(MESSAGE_INVALID_COMMAND_FORMAT,
+                        DelMedConCommand.MESSAGE_USAGE));
+            } else if (medConStr.length() > 45) {
```
Comment (yadobler) [inline src/main/java/seedu/address/logic/parser/DelMedConCommandParser.java:58]: Good guard clauses

File: src/test/java/seedu/address/logic/parser/AddMedConCommandParserTest.java [hunk 3]
```diff
@@ -73,6 +74,12 @@ public void parse_compulsoryFieldsMissing_failure() {
         assertParseFailure(parser, MEDCON_DESC_AMY, String.format(MESSAGE_INVALID_COMMAND_FORMAT,
                 AddMedConCommand.MESSAGE_USAGE));
 
+        // MedCon parameter missing
+        assertParseFailure(parser, NRIC_DESC_AMY, String.format(MESSAGE_INVALID_COMMAND_FORMAT,
+                AddMedConCommand.MESSAGE_USAGE));
+        assertParseFailure(parser, NRIC_DESC_AMY + " " + PREFIX_MEDCON,
+                String.format(MESSAGE_INVALID_COMMAND_FORMAT, AddMedConCommand.MESSAGE_USAGE));
+
```
Comment (yadobler) [inline src/test/java/seedu/address/logic/parser/AddMedConCommandParserTest.java:82]: Good comprehensive testing

**Review by yadobler — APPROVED** [review]
lgtm

File: src/main/java/seedu/address/logic/parser/AddMedConCommandParser.java [hunk 4]
```diff
@@ -45,10 +45,14 @@ public AddMedConCommand parse(String args) throws ParseException {
         // Parse all MedCon values and add them to a set
         Set<MedCon> medCons = new HashSet<>();
         for (String medConStr : argMultimap.getAllValues(PREFIX_MEDCON)) {
-            if (medConStr.length() > 45) {
+            if (medConStr.isEmpty()) {
+                throw new ParseException(String.format(MESSAGE_INVALID_COMMAND_FORMAT,
+                        AddMedConCommand.MESSAGE_USAGE));
+            } else if (medConStr.length() > 45) {
```
Comment (yong-zhuo) [inline src/main/java/seedu/address/logic/parser/AddMedConCommandParser.java:51]: Good check for empty string

**Review by yong-zhuo — APPROVED** [review]
Overall LGTM

**Pull request merged**